app/controllers/concerns/paginator.rb

```diff
@@ -0,0 +1,49 @@
+class Paginator
```

Comment:
Maybe this class should be moved to `lib`. It is used on controllers but it is not a concern related just to the controller layer. Also, there is already a lot of logic here to justify unit tests for it.